src/surface_tracker/location.py

```diff
@@ -298,6 +298,17 @@ def __map_points(
 ##### Helper functions
 
 
+def _perspective_transform(points, transform_matrix):
```

Comment:
I think, you can get rid of the for loop by defining this function (I think this is also changed later in this PR, right?) as:

```
def _perspective_transform(points, transform_matrix):
    homogeneous_points = cv2.convertPointsToHomogeneous(points)[:,0,:]
    projected_points = (transform_matrix @ homogeneous_points.T).T
    projected_points[:,2] = np.abs(projected_points[:,2])
    return cv2.convertPointsFromHomogeneous(projected_points)[:,0,:]

```